cola/widgets/selectcommits.py

```diff
@@ -135,6 +143,13 @@ def commit_oid_double_clicked(self, _item):
         if oid:
             self.accept()
 
+    def search_list(self, text):
+        all_items = self.commits.findItems('.*', Qt.MatchRegExp)
+        search_items = self.commits.findItems(text, Qt.MatchContains)
+        for items in all_items:
+            items.setHidden(True)
+        for items in search_items:
+            items.setHidden(False)
```

Comment:
This doesn't look very efficient... What's the purpose here? Hide all items first, then unhide the search results? Why create a search that finds all items in the first place? Wouldn't `for items in self.commits` just work (tm)?

Reply:
Hi yes i think that would be better let me try that.

Reply:
```
search_items = self.commits.findItems(text, Qt.MatchContains)
for i in range(self.commits.count()):
  self.commits.item(i).setHidden(True)
for items in search_items:
  items.setHidden(False)
```
This works.

I can't use a 
for items in self.commits.items loop as self.commits.items is not iterable.

Reply:
Ah okay. I don't know the internal list APIs of Python (or Qt) very well, I think there's probably some function to return an enumerable type from `self.commits` that can be passed into the for loop. I'm not sure how efficient lookup by position is (`item(i)`, is it O(1)?). But looking at Qt QListWidget examples, this seems to be the correct way to go. So: LGTM

Reply:
Yes ! Should i resolve this conversation ? What should i do regarding failing checks?

Reply:
It fails in line 153 where it expects two blank lines instead of just one:
`/home/travis/build/git-cola/git-cola/cola/widgets/selectcommits.py:153:1: E302 expected 2 blank lines, found 1`

That's probably just a code style issue (not a functional issue) but you should resolve it.

Reply:
Probably there should be two blank lines preceding a class definition for readability reasons...